torch_geometric_temporal/nn/convolutional/mtgnn.py

```diff
@@ -0,0 +1,445 @@
+                if self._seq_length > self._receptive_field:
+                    self._skip_convs.append(nn.Conv2d(in_channels=conv_channels,
+                                                     out_channels=skip_channels,
+                                                     kernel_size=(1, self._seq_length-rf_size_j+1)))
+                else:
+                    self._skip_convs.append(nn.Conv2d(in_channels=conv_channels,
+                                                     out_channels=skip_channels,
+                                                     kernel_size=(1, self._receptive_field-rf_size_j+1)))
+
+                if self._gcn_true:
+                    self._mixprop_conv1.append(
+                        MixProp(conv_channels, residual_channels, gcn_depth, dropout, propalpha))
+                    self._mixprop_conv2.append(
+                        MixProp(conv_channels, residual_channels, gcn_depth, dropout, propalpha))
+
+                if self._seq_length > self._receptive_field:
+                    self._normlization.append(LayerNorm(
+                        (residual_channels, num_nodes, self._seq_length - rf_size_j + 1), elementwise_affine=layer_norm_affline))
+                else:
+                    self._normlization.append(LayerNorm(
+                        (residual_channels, num_nodes, self._receptive_field - rf_size_j + 1), elementwise_affine=layer_norm_affline))
+
+                new_dilation *= dilation_exponential
+
+        self._layers = layers
+        self._end_conv_1 = nn.Conv2d(in_channels=skip_channels,
+                                    out_channels=end_channels,
+                                    kernel_size=(1, 1),
+                                    bias=True)
+        self._end_conv_2 = nn.Conv2d(in_channels=end_channels,
+                                    out_channels=out_dim,
+                                    kernel_size=(1, 1),
+                                    bias=True)
+        if self._seq_length > self._receptive_field:
+            self._skip_conv_0 = nn.Conv2d(in_channels=in_dim, out_channels=skip_channels, kernel_size=(
+                1, self._seq_length), bias=True)
+            self._skip_conv_E = nn.Conv2d(in_channels=residual_channels, out_channels=skip_channels, kernel_size=(
+                1, self._seq_length-self._receptive_field+1), bias=True)
+
+        else:
+            self._skip_conv_0 = nn.Conv2d(in_channels=in_dim, out_channels=skip_channels, kernel_size=(
+                1, self._receptive_field), bias=True)
+            self._skip_conv_E = nn.Conv2d(in_channels=residual_channels,
+                                   out_channels=skip_channels, kernel_size=(1, 1), bias=True)
+
+        self._idx = torch.arange(self._num_nodes)
+
+    def forward(self, X_in: torch.FloatTensor, predefined_A: Optional[torch.FloatTensor]=None, idx: Optional[torch.LongTensor]=None, static_feat: Optional[torch.FloatTensor]=None) -> torch.FloatTensor:
+        """
```

Comment:
Have smaller methods.

Reply:
What do you mean by "smaller methods"?

Reply:
Not a single forward with 50+ lines. Have smaller functional units with abstractions.